canal/sync.go

```diff
@@ -222,6 +238,22 @@ func (c *Canal) GetMasterPos() (mysql.Position, error) {
 	return mysql.Position{name, uint32(pos)}, nil
 }
 
+func (c *Canal) GetMasterGTIDSet() (mysql.GTIDSet, error) {
+	rr, err := c.Execute("SELECT @@GLOBAL.GTID_EXECUTED")
```

Comment:
the query is just for MySQL, can it fit for MariaDB?

Reply:
For MariaDB the query should be `SELECT @@GLOBAL.gtid_current_pos`.
(https://mariadb.com/kb/en/library/gtid/#gtid_current_pos)